Common/NearestNeighbour/src/main/java/org/tribuo/common/nearest/KNNTrainer.java

```diff
@@ -44,25 +47,42 @@
 
     /**
      * The available distance functions.
+     * @deprecated
+     * This Enum is deprecated in version 4.3, replaced by {@link DistanceType}
      */
+    @Deprecated
     public enum Distance {
         /**
          * L1 (or Manhattan) distance.
          */
-        L1,
+        L1(DistanceType.L1),
         /**
          * L2 (or Euclidean) distance.
          */
-        L2,
+        L2(DistanceType.L2),
         /**
          * Cosine similarity used as a distance measure.
          */
-        COSINE
+        COSINE(DistanceType.COSINE);
+
+        private final DistanceType distanceType;
+
+        Distance(DistanceType distanceType) {
+            this.distanceType = distanceType;
+        }
+
+        public DistanceType getDistanceType() {
+            return distanceType;
+        }
     }
 
-    @Config(mandatory = true, description="The distance function used to measure nearest neighbours.")
+    @Deprecated
+    @Config(mandatory = true, description="The distance function used to measure nearest neighbours. This is now deprecated.")
```

Comment:
Both this and `distType` can't be mandatory as it'll crash in postConfig. I think neither of them should be mandatory, like in `KMeansTrainer`.

Reply:
Good catch on that one.